modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/core/schema/DataDefinition.es.js

```diff
@@ -50,22 +50,14 @@ export class DataDefinitionSchema extends Schema {
 			return this[SYMBOL_CACHE].dataDefinitionFields;
 		}
 		else {
-			const fields = [...dataDefinition.dataDefinitionFields];
+			const fields = [];
 			const visitor = new PagesVisitor(pages);
 
 			visitor.mapFields((field) => {
-				const index = fields.findIndex(
-					({name}) => name === field.fieldName
+				const dataDefinitionField = DataConverter.getDataDefinitionField(
+					field
 				);
-
-				const newField = DataConverter.getDataDefinitionField(field);
-
-				if (index === -1) {
-					fields.push(newField);
-				}
-				else {
-					fields[index] = newField;
-				}
+				fields.push(dataDefinitionField);
```

Comment:
This change will break the App Builder, remember that the `dataDefinitionFields` can be different from the `Pages` fields, We always need to take into account what is inside the `dataDefinitionFields` and then merge with the fields within `Pages`, that is, there are global fields and fields at form level, this is the behavior of `dataDefinitionFields`.

They are comparable but there are different behaviors, here I also dealt with the case that the same field already exists within `Pages`. If you still have doubts about the difference between `dataLayoutFields`, `dataDefinitionFields`, and `Pages` see Adriano or some backend to explain how this works.